Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm

```diff
@@ -862,24 +862,34 @@ static void prewarmLogs()
     if (!connectionPair)
         CRASH();
     auto [streamConnection, serverHandle] = WTFMove(*connectionPair);
-    m_logStreamConnection = WTFMove(streamConnection);
-    if (RefPtr logStreamConnection = m_logStreamConnection)
-        logStreamConnection->open(*this);
 
-    parentProcessConnection()->sendWithAsyncReply(Messages::WebProcessProxy::SetupLogStream(getpid(), WTFMove(serverHandle), WebProcess::singleton().m_logStreamIdentifier), [] (IPC::Semaphore&& wakeUpSemaphore, IPC::Semaphore&& clientWaitSemaphore) {
-        if (RefPtr logStreamConnection = WebProcess::singleton().m_logStreamConnection)
-            logStreamConnection->setSemaphores(WTFMove(wakeUpSemaphore), WTFMove(clientWaitSemaphore));
+    LogStreamIdentifier logStreamIdentifier { LogStreamIdentifier::generate() };
+
+    RefPtr logStreamConnection = WTFMove(streamConnection);
+    if (!logStreamConnection)
+        return;
+
+    logStreamConnection->open(*this, RunLoop::protectedCurrent());
+
+    parentProcessConnection()->sendWithAsyncReply(Messages::WebProcessProxy::SetupLogStream(getpid(), WTFMove(serverHandle), logStreamIdentifier), [logStreamConnection, logStreamIdentifier] (IPC::Semaphore&& wakeUpSemaphore, IPC::Semaphore&& clientWaitSemaphore) {
+        logStreamConnection->setSemaphores(WTFMove(wakeUpSemaphore), WTFMove(clientWaitSemaphore));
 #if PLATFORM(IOS_FAMILY)
         prewarmLogs();
 #endif
+        webkitLogClient() = WTF::makeUnique<WebKitLogClient>(*logStreamConnection.get(), logStreamIdentifier);
+        webCoreLogClient() = WTF::makeUnique<WebKitLogClient>(*logStreamConnection.get(), logStreamIdentifier);
```

Comment:
ditto.

Reply:
Maybe WebKitLogClient should be ref-counted and we could use the same instance for `webCoreLogClient()` and `webkitLogClient()`? Seems a bit wasteful to allocate 2 objects. Or we could keep using unique_ptr but have a single `logClient()` in WebCore, used by both WebKit and WebCore?